### PR TITLE
Support date timezone

### PR DESCRIPTION
**Pull request by kepricon**
### What this PR does
- support { date: <dateExpression>, timezone: <tzExpression> } format for date_operators such as [$week](https://www.mongodb.com/docs/manual/reference/operator/aggregation/week/), [$year](https://www.mongodb.com/docs/manual/reference/operator/aggregation/year/)
### related ticket
https://github.com/mongomock/mongomock/issues/783

### CODE REVIEW

File: mongomock/aggregate.py [hunk 1]
```diff
@@ -576,7 +577,14 @@ def _handle_string_operator(self, operator, values):
             'pipeline, it is currently not implemented  in Mongomock.' % operator)
 
     def _handle_date_operator(self, operator, values):
-        out_value = self.parse(values)
+        if isinstance(values, dict) and \
```
Comment (pcorpet) [inline mongomock/aggregate.py]: I suggest you use a simpler set equality: `values.keys() == {'date', 'timezone'}`

**Review by pcorpet — APPROVED** [review]
Thanks a lot for this PR. Please address my few comments before I merge it.

File: mongomock/aggregate.py (hunk 1 above)
Comment (kepricon) [reply mongomock/aggregate.py]: thank you.
I updated it as you suggested.

**Comment by kepricon** [comment]
@pcorpet thank you for the review.
I updated the PR.

when is the next release?

**Comment by pcorpet** [comment]
There is no regular release and I just did one last week. I'd rather wait August. Ping me again in a few weeks if you wish.

**Pull request merged**